packages/instrumentation-openai/src/instrumentation.ts

```diff
@@ -0,0 +1,748 @@
+
+        let startInfo;
+        try {
+          startInfo = self._startEmbeddingsSpan(params, this?._client?.baseURL);
+        } catch (err) {
+          self._diag.error('unexpected error starting span:', err);
+          return original.apply(this, args);
+        }
+        const { span, ctx, commonAttrs } = startInfo;
+
+        /** @type {import('openai/core').APIPromise} */
+        const apiPromise = context.with(ctx, () => original.apply(this, args));
+
+        apiPromise
+          .then((result: any) => {
+            self._onEmbeddingsCreateResult(span, startNow, commonAttrs, result);
+          })
+          .catch(
+            self._createAPIPromiseRejectionHandler(startNow, span, commonAttrs)
+          );
+
+        return apiPromise;
+      };
+    };
+  }
+
+  /**
+   * Start a span for this chat-completion API call. This also emits log events
+   * as appropriate for the request params.
+   *
+   * @param {OpenAIInstrumentationConfig} config
+   */
+  _startEmbeddingsSpan(params: any, baseURL: string | undefined) {
+    // Attributes common to span, metrics, log events.
+    const commonAttrs: Attributes = {
+      [ATTR_GEN_AI_OPERATION_NAME]: 'embeddings',
+      [ATTR_GEN_AI_REQUEST_MODEL]: params.model,
+      [ATTR_GEN_AI_SYSTEM]: 'openai',
+    };
+    Object.assign(commonAttrs, getAttrsFromBaseURL(baseURL, this._diag));
+
+    // Span attributes.
+    const attrs: Attributes = {
+      ...commonAttrs,
+    };
+    if (params.encoding_format != null) {
+      attrs[ATTR_GEN_AI_REQUEST_ENCODING_FORMATS] = [params.encoding_format];
+    }
+
+    const span = this.tracer.startSpan(
+      `${attrs[ATTR_GEN_AI_OPERATION_NAME]} ${attrs[ATTR_GEN_AI_REQUEST_MODEL]}`,
+      {
+        kind: SpanKind.CLIENT,
+        attributes: attrs,
+      }
+    );
+    const ctx = trace.setSpan(context.active(), span);
+
+    return { span, ctx, commonAttrs };
+  }
+
+  /**
+   * @param {import('openai').OpenAI.CreateEmbeddingResponse} result
+   */
+  _onEmbeddingsCreateResult(
+    span: Span,
+    startNow: number,
+    commonAttrs: Attributes,
+    result: any
+  ) {
+    debug('OpenAI.Embeddings.create result: %O', result);
+    try {
+      span.setAttribute(ATTR_GEN_AI_RESPONSE_MODEL, result.model);
+
+      this._genaiClientOperationDuration.record(
+        (performance.now() - startNow) / 1000,
+        {
+          ...commonAttrs,
+          [ATTR_GEN_AI_RESPONSE_MODEL]: result.model,
+        }
+      );
+
+      span.setAttribute(
+        ATTR_GEN_AI_USAGE_INPUT_TOKENS,
+        result.usage.prompt_tokens
+      );
+      this._genaiClientTokenUsage.record(result.usage.prompt_tokens, {
+        ...commonAttrs,
+        [ATTR_GEN_AI_RESPONSE_MODEL]: result.model,
+        [ATTR_GEN_AI_TOKEN_TYPE]: 'input',
+      });
+    } catch (err) {
+      this._diag.error(
+        'unexpected error getting telemetry from embeddings result:',
+        err
+      );
+    }
+    span.end();
+  }
+}
+
+module.exports = {
```

Comment:
TS compile spec will generate the exports (in its own way). Since you have to export the class because of types this can be removed

Reply:
Yup, thanks. Commit efb119cd96b3040cdbb12fcf132389688a4eafb8 as well.